### PR TITLE
Feat/2583 edit save set tag

### PR DESCRIPTION
**Pull request by adipaul1981**

### CODE REVIEW

File: src/components/CohortBuilder/ParticipantsTableView/ParticipantsTable.js [hunk 1]
```diff
@@ -253,6 +254,7 @@ class ParticipantsTable extends Component {
                 showModal: false,
               })
             }
+            saveSetActionType={'create'}
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/ParticipantsTableView/ParticipantsTable.js:257]: I would use ENUM

File: src/store/actionCreators/Test/saveSets.test.tsx [hunk 2]
```diff
@@ -18,7 +22,15 @@ import {
...
-import { deleteSaveSet, getSetAndParticipantsCountByUser, saveSetCountForTag } from 'services/sets';
+import {
+  deleteSaveSet,
+  editSaveSetTag,
+  getSetAndParticipantsCountByUser,
+  saveSetCountForTag,
+} from 'services/sets';
+// @ts-ignore
+import saveSet from '@kfarranger/components/dist/utils/saveSet';
+import { SaveSetInfo } from '../../../components/UserDashboard/ParticipantSets';
```
Comment (evans-g-crsj) [inline src/store/actionCreators/Test/saveSets.test.tsx:33]: remove ../...

File: src/store/actionCreators/saveSets.ts [hunk 3]
```diff
@@ -13,16 +16,23 @@ import {
...
+  editSaveSetTag,
+  getSetAndParticipantsCountByUser,
+  saveSetCountForTag,
+} from 'services/sets';
 // @ts-ignore
 import saveSet from '@kfarranger/components/dist/utils/saveSet';
 import { RootState } from '../rootState';
 import graphql from 'services/arranger';
+import { initializeApi } from 'services/api';
+import { SaveSetInfo } from '../../components/UserDashboard/ParticipantSets';
```
Comment (evans-g-crsj) [inline src/store/actionCreators/saveSets.ts:30]: ../../

File: src/store/actionCreators/saveSets.ts [hunk 4]
```diff
@@ -72,6 +81,36 @@ export const createSaveSetIfUnique = (
...
 
+export const editSaveSet = (
+  payload: EditSetParams,
+): ThunkAction<void, RootState, null, SaveSetsActionTypes> => async (dispatch) => {
+  const { saveSetInfo, onNameConflict, onSuccess, onFail } = payload;
+
+  try {
+    const tagIsUnique = (await saveSetCountForTag(saveSetInfo.name, saveSetInfo.currentUser)) === 0;
+    if (tagIsUnique) {
+      const result = await editSaveSetTag(saveSetInfo);
```
Comment (evans-g-crsj) [inline src/store/actionCreators/saveSets.ts:92]: result is a bit vague for me in the sense that I don t know what I should expect for result. Maybe you could use a more meaningful name for instance nOfSaveSetEdited or something in that sense.

File: src/store/actionCreators/saveSets.ts [hunk 5]
```diff
@@ -72,6 +81,36 @@ export const createSaveSetIfUnique = (
...
+      }
+    }
+
+    if (onNameConflict) {
+      onNameConflict();
+    }
+
+    dispatch(failureCreate(new SaveSetNameConflictError('A set with this name already exists')));
+  } catch (error) {
+    console.error(error);
```
Comment (evans-g-crsj) [inline src/store/actionCreators/saveSets.ts:110]: Just to make sure => nothing to clean here?

File: src/store/reducers/Test/saveSets.test.tsx [hunk 6]
```diff
@@ -6,7 +6,9 @@ import {
   isLoadingCreateSaveSet,
   isLoadingSaveSets,
   isDeletingSaveSets,
+  isEditingTag,
 } from '../../actionCreators/saveSets';
+import { SaveSetInfo } from '../../../components/UserDashboard/ParticipantSets';
```
Comment (evans-g-crsj) [inline src/store/reducers/Test/saveSets.test.tsx:11]: ../...

File: src/store/actionCreators/saveSets.ts (hunk 5 above)
Comment (adipaul1981) [reply src/store/actionCreators/saveSets.ts:110]: no. the edit failed thats all

**Pull request closed without merging**